packages/aws-cdk/test/api/hotswap-deployments.test.ts

```diff
@@ -1,343 +1,185 @@
-import * as cxapi from '@aws-cdk/cx-api';
-import { CloudFormation, Lambda } from 'aws-sdk';
+import { Lambda, StepFunctions } from 'aws-sdk';
 import { tryHotswapDeployment } from '../../lib/api/hotswap-deployments';
-import { testStack, TestStackArtifact } from '../util';
-import { MockSdkProvider } from '../util/mock-sdk';
-import { FakeCloudformationStack } from './fake-cloudformation-stack';
+import * as setup from './hotswap-test-setup';
 
-const STACK_NAME = 'withouterrors';
-const STACK_ID = 'stackId';
-
-let mockSdkProvider: MockSdkProvider;
 let mockUpdateLambdaCode: (params: Lambda.Types.UpdateFunctionCodeRequest) => Lambda.Types.FunctionConfiguration;
-let currentCfnStack: FakeCloudformationStack;
-const currentCfnStackResources: CloudFormation.StackResourceSummary[] = [];
+let mockUpdateMachineDefinition: (params: StepFunctions.Types.UpdateStateMachineInput) => StepFunctions.Types.UpdateStateMachineOutput;
 
 beforeEach(() => {
-  jest.resetAllMocks();
-  mockSdkProvider = new MockSdkProvider({ realSdk: false });
+  setup.setupHotswapTests();
   mockUpdateLambdaCode = jest.fn();
-  mockSdkProvider.stubLambda({
+  setup.mockSdkProvider.stubLambda({
     updateFunctionCode: mockUpdateLambdaCode,
   });
```

Comment:
Since we now have a separate API for this in `setup`, how about we make this a little bit nicer, so I can say

```suggestion
setup.setUpdateFunctionCodeMock(mockUpdateLambdaCode);
```

?